pallets/connectors-gateway/src/tests.rs

```diff
@@ -0,0 +1,480 @@
+				Error::<Runtime>::DomainNotSupported
+			);
+
+			let storage_entry = DomainRouters::<Runtime>::get(domain);
+			assert!(storage_entry.is_none());
+		});
+	}
+}
+
+mod add_connector {
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(ConnectorsGateway::add_connector(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert!(ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+
+			event_exists(Event::<Runtime>::ConnectorAdded {
+				connector: domain_address,
+			});
+		});
+	}
+
+	#[test]
+	fn bad_origin() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_noop!(
+				ConnectorsGateway::add_connector(
+					RuntimeOrigin::signed(get_random_test_account_id()),
+					domain_address.clone(),
+				),
+				BadOrigin
+			);
+
+			assert!(!ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+		});
+	}
+
+	#[test]
+	fn unsupported_domain() {
+		new_test_ext().execute_with(|| {
+			let domain_address = DomainAddress::Centrifuge(get_random_test_account_id().into());
+
+			assert_noop!(
+				ConnectorsGateway::add_connector(RuntimeOrigin::root(), domain_address.clone()),
+				Error::<Runtime>::DomainNotSupported
+			);
+
+			assert!(!ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+		});
+	}
+
+	#[test]
+	fn connector_already_added() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(ConnectorsGateway::add_connector(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert!(ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+
+			assert_noop!(
+				ConnectorsGateway::add_connector(RuntimeOrigin::root(), domain_address,),
+				Error::<Runtime>::ConnectorAlreadyAdded
+			);
+		});
+	}
+}
+
+mod remove_connector {
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let address = H160::from_slice(&get_random_test_account_id().as_slice()[..20]);
+			let domain_address = DomainAddress::EVM(0, address.into());
+
+			assert_ok!(ConnectorsGateway::add_connector(
+				RuntimeOrigin::root(),
+				domain_address.clone(),
+			));
+
+			assert!(ConnectorsAllowlist::<Runtime>::contains_key(
+				domain_address.domain(),
+				domain_address.clone()
+			));
+
+			event_exists(Event::<Runtime>::ConnectorAdded {
+				connector: domain_address.clone(),
+			});
```

Comment:
Nit: We are already asserting these steps above in `mod add_connector`.

Reply:
I'll remove it, this is test paranoia that built over time =D